### ISSUE TITLE
TF:2.0 Webcam Demo ?

### ISSUE THREAD

**Issue by pwnmactep** [googlecreativelab/teachablemachine-community#55]
Hello ,

heave/can somebody give a working TF:2.0 Webcam Demo Code  where i can load my Tensorflow (Keras) model  created by [https://teachablemachine.withgoogle.com/train/image](url) (not Tensorflow.js)

Would be very Helpfull.

**Comment by GautamBose** [comment]
Hi there, 

I created this short snippet for you that uses the webcam in combination with a downloaded keras model. Its been tested with tf 2.0 as well. It does require opencv which can be installed as follows: 

`pip3 install opencv-python`

Here is the snippet

`
import cv2 # import opencv
import tensorflow.keras as keras
import numpy as np

webcam = cv2.VideoCapture(0)
model = keras.models.load_model('keras_model.h5')

data_for_model = np.ndarray(shape=(1, 224, 224, 3), dtype=np.float32)

# This function proportionally resizes the image from your webcam to 224 pixels high
def image_resize(image, width = None, height = None, inter = cv2.INTER_AREA):
    dim = None
    (h, w) = image.shape[:2]
    if width is None and height is None:
        return image
    if width is None:
        r = height / float(h)
        dim = (int(w * r), height)
    else:
        r = width / float(w)
        dim = (width, int(h * r))
    resized = cv2.resize(image, dim, interpolation = inter)
    return resized

# this function crops to the center of the resize image
def cropTo(img):
    size = 224
    height, width = img.shape[:2]

    sideCrop = (width - 224) // 2
    return img[:,sideCrop:(width - sideCrop)]

while True:
    ret, img = webcam.read()
    if ret:
        #same as the cropping process in TM2
        img = image_resize(img, height=224)
        img = cropTo(img)

        # flips the image
        img = cv2.flip(img, 1)

        #normalize the image and load it into an array that is the right format for keras
        normalized_img = (img.astype(np.float32) / 127.0) - 1
        data_for_model[0] = normalized_img
 
        #run inference
        prediction = model.predict(data_for_model)

        print(prediction)

        cv2.imshow('webcam', img)
        if cv2.waitKey(1) == 27:
            break

cv2.destroyAllWindows()
`

**Comment by pwnmactep** [comment]
Hi , 
You should add this snippet / example to  [ https://teachablemachine.withgoogle.com/train/image](url)  for all. 

Thank you very much :)

**Comment by pwnmactep** [comment]
Hello ,It works Thank you. How can i use the labels.txt to show the Results by name ?

**Comment by GautamBose** [comment]
Hi

I've added a 'load_labels' function that you can use to load the labels.txt into a python list, and a small change to the way the confidence is printed to include the labels.

python`
import cv2 # import opencv
import tensorflow.keras as keras
import numpy as np

np.set_printoptions(suppress=True)

webcam = cv2.VideoCapture(0)
model = keras.models.load_model('keras_model.h5')
data_for_model = np.ndarray(shape=(1, 224, 224, 3), dtype=np.float32)

def load_labels(path):
	f = open(path, 'r')
	lines = f.readlines()
	labels = []
	for line in lines:
		labels.append(line.split(' ')[1].strip('\n'))
	return labels

label_path = 'labels.txt'
labels = load_labels(label_path)
print(labels)

# This function proportionally resizes the image from your webcam to 224 pixels high
def image_resize(image, height, inter = cv2.INTER_AREA):
    dim = None
    (h, w) = image.shape[:2]
    r = height / float(h)
    dim = (int(w * r), height)
    resized = cv2.resize(image, dim, interpolation = inter)
    return resized

# this function crops to the center of the resize image
def cropTo(img):
    size = 224
    height, width = img.shape[:2]

    sideCrop = (width - 224) // 2
    return img[:,sideCrop:(width - sideCrop)]

while True:
    ret, img = webcam.read()
    if ret:
        #same as the cropping process in TM2
        img = image_resize(img, height=224)
        img = cropTo(img)

        # flips the image
        img = cv2.flip(img, 1)

        #normalize the image and load it into an array that is the right format for keras
        normalized_img = (img.astype(np.float32) / 127.0) - 1
        data_for_model[0] = normalized_img
 
        #run inference
        prediction = model.predict(data_for_model)
        for i in range(0, len(prediction[0])):
            print('{}: {}'.format(labels[i], prediction[0][i]))
        cv2.imshow('webcam', img)
        if cv2.waitKey(1) == 27:
            break

cv2.destroyAllWindows()
`

**Comment by pwnmactep** [comment]
Hello,

thank you very much , that helps a lot :)

Wish you a great Time !
